cmd/podman/containers/ps.go

```diff
@@ -440,22 +517,27 @@ func comparePorts(i, j ocicni.PortMapping) bool {
 	return i.Protocol < j.Protocol
 }
 
-// formatGroup returns the group as <IP:startPort:lastPort->startPort:lastPort/Proto>
-// e.g 0.0.0.0:1000-1006->1000-1006/tcp.
-func formatGroup(key string, start, last int32) string {
+// formatGroup returns the group in the format:
+// <IP:firstHost:lastHost->firstCtr:lastCtr/Proto>
+// e.g 0.0.0.0:1000-1006->2000-2006/tcp.
+func formatGroup(key string, firstHost, lastHost, firstCtr, lastCtr int32) string {
 	parts := strings.Split(key, "/")
 	groupType := parts[0]
 	var ip string
 	if len(parts) > 1 {
 		ip = parts[0]
 		groupType = parts[1]
 	}
-	group := strconv.Itoa(int(start))
-	if start != last {
-		group = fmt.Sprintf("%s-%d", group, last)
-	}
-	if ip != "" {
-		group = fmt.Sprintf("%s:%s->%s", ip, group, group)
+
+	group := func(first, last int32) string {
+		group := strconv.Itoa(int(first))
+		if first != last {
+			group = fmt.Sprintf("%s-%d", group, last)
+		}
+		return group
 	}
-	return fmt.Sprintf("%s/%s", group, groupType)
+	hostGroup := group(firstHost, lastHost)
+	ctrGroup := group(firstCtr, lastCtr)
+
+	return fmt.Sprintf("%s:%s->%s/%s", ip, hostGroup, ctrGroup, groupType)
```

Comment:
I think this bit handles an empty IP incorrectly - looks like it will still print the `:` after it even if there is no IP set.

Reply:
We only pass the `fmt.Sprintf("%s/%s", hostIP, first.Protocol)` to this function (line 496), which contains the `hostIP`. The `hostIP` is pre-checked for an empty string and still set to `0.0.0.0` (477-480).